Studio: Add missing tooltip labels to buttons

ivan-ottinger opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

Fixes https://github.com/Automattic/dotcom-forge/issues/6623.

## Proposed Changes

- add new tooltip labels for the following buttons: `Open settings`, `Start site`, `Starting` & `Stop site`, "site screenshot" and "copy to clipboard" buttons
- update aria-label for the following buttons: `Account` →  `Open settings`; `Profile link` → `Edit profile` & `Help` → `Get help`

| Start site | Stop site |
|--------|--------|
| ![Markup on 2024-11-08 at 16:42:40](https://github.com/user-attachments/assets/f405e71c-0fa5-4f68-9235-08723e309390) | ![Markup on 2024-11-08 at 16:42:11](https://github.com/user-attachments/assets/24724026-676d-45c9-a922-8a0fc46360a9) | 

| Open settings | Site screenshot |
|--------|--------|
| ![Markup on 2024-11-11 at 16:50:54](https://github.com/user-attachments/assets/256aad8f-6f37-465b-99c2-cf448b914ef3) | ![Markup on 2024-11-11 at 16:49:31](https://github.com/user-attachments/assets/a0bcf072-e3e7-423d-a46c-29c2c04756b5) | 

| Toggle sidebar | Copy to clipboard |
|--------|--------|
| ![Markup on 2024-11-11 at 16:52:33](https://github.com/user-attachments/assets/643d4751-2191-439c-887a-6b250d6bbe19) | ![Markup on 2024-11-11 at 16:52:04](https://github.com/user-attachments/assets/1aa88c98-8008-45fc-82af-1da693733bc8) | 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Check out the PR and build the Studio app.
2. Navigate through the app and hover over the buttons where the tooltip labels have been added.
3. Tooltip labels should display correctly.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Have you checked for TypeScript, React or other console errors?

ivan-ottinger committed:
Add `Edit Profile` button tooltip label

ivan-ottinger committed:
Update `aria-label` of the `Edit Profile` button

ivan-ottinger committed:
Add `Open Settings` button tooltip label

ivan-ottinger committed:
Update `aria-label` of the `Open Settings` button

ivan-ottinger committed:
Update `aria-label` of the `Get help` button

ivan-ottinger committed:
Add `Get help` button tooltip label

ivan-ottinger committed:
Add `Stop site` / `Start site` tooltip label to the Stop / Start site button

ivan-ottinger committed:
Fix unit tests

ivan-ottinger committed:
Fix linting

katinthehatsite commented on src/components/user-settings.tsx:
```diff
@@ -33,7 +33,9 @@ const UserInfo = ( {
 					onClick={ () => getIpcApi().openURL( WPCOM_PROFILE_URL ) }
-					aria-label={ __( 'Profile link' ) }
+					aria-label={ __( 'Edit Profile' ) }
+					label={ __( 'Edit Profile' ) }
```
Not sure if it should be `Edit profile` or `Edit Profile` considering that before it was `Profile link`?

wojtekn replied:
Good point, let's stick with the 'Edit profile' casing in all cases.

ivan-ottinger replied:
I was considering both options but then picked `Edit Profile` and `Open Settings`. Regardless, using `Edit profile` and `Open settings` sounds good to me as well, so I will apply the changes as you suggested. 🙂

ivan-ottinger replied:
I have now addressed the change in https://github.com/Automattic/studio/pull/648/commits/791c4b0c3a290d2414990ceebdea212a3cc792f0.

katinthehatsite reviewed (commented):
The changes look good to me and they worked as expected.

One thing is that I am not sure whether we should also add a tooltip for toggling the sidebar? What do you think?

<img width="448" alt="Screenshot 2024-11-11 at 10 51 46 AM" src="https://github.com/user-attachments/assets/134e4386-09ac-467b-81f8-2efe5ae5f56d">

Another small thing that I noticed is that when the site is starting (e.g. the green indicator is blinking), the tooltip still says `Start site`. Should we remove when this process is happening or display "Starting" instead?

wojtekn commented:
In addition to the sidebar toggle mentioned by Kat, I would propose to add a few more:
- full WP Admin and site URLs when the user hovers over 'WP admin', 'Open site', or site screenshot
- 'Copy' when a user hovers over links that can be copied to the clipboard in the Settings tab
- settings link displayed for a user who is logged out

Also, do you think we should add tooltips for items that are already somehow descriptive e.g. tabs, buttons like 'Add site' or 'Export entire site' etc.?

ivan-ottinger committed:
Change `Edit Profile` to `Edit profile` and `Open Settings` to `Open settings`

ivan-ottinger committed:
Add `Toggle sidebar` tooltip

ivan-ottinger committed:
Add `Starting` tooltip text when a site is starting

ivan-ottinger committed:
Add tooltip handling to the `CopyTextButton` component

ivan-ottinger committed:
Add tooltips to "Copy to clipboard" buttons on the site Settings page

ivan-ottinger committed:
Add tooltips to `WP admin` and `Open site` links

ivan-ottinger committed:
Add tooltip to the site screenshot on the Overview tab

ivan-ottinger committed:
Fix linting errors

ivan-ottinger committed:
Fix translators comments

ivan-ottinger committed:
Update "Copy to clipboard" tooltip logic

ivan-ottinger committed:
Fix unit tests

ivan-ottinger commented:
Thank you for your review and feedback, @katinthehatsite and @wojtekn! 🙂 

> One thing is that I am not sure whether we should also add a tooltip for toggling the sidebar? What do you think?

Good point! I have added tooltip there as well.

![Markup on 2024-11-11 at 16:52:33](https://github.com/user-attachments/assets/6f72ab17-cad5-466e-a8f7-51d265a6ba2d)

> Another small thing that I noticed is that when the site is starting (e.g. the green indicator is blinking), the tooltip still says Start site. Should we remove when this process is happening or display "Starting" instead?

Nice catch! I have added `Starting` tooltip when the site is starting.

> - full WP Admin and site URLs when the user hovers over 'WP admin', 'Open site', or site screenshot

Done:

![Markup on 2024-11-11 at 16:49:52](https://github.com/user-attachments/assets/c7d08e65-e2b9-474d-8497-90d3e7d688dd)

![Markup on 2024-11-11 at 16:50:21](https://github.com/user-attachments/assets/a4bd794c-2db2-4227-82c8-988bdc47b8c8)

> - 'Copy' when a user hovers over links that can be copied to the clipboard in the Settings tab

Done:

![Markup on 2024-11-11 at 16:52:04](https://github.com/user-attachments/assets/b1b8e295-090d-4f6c-944e-fe1fd7c0a164)
(example showing tooltip of one of the buttons)

> - settings link displayed for a user who is logged out

Are you referring to the following `Log in` button?

![Markup on 2024-11-11 at 15:56:34](https://github.com/user-attachments/assets/b89e8c92-e444-4517-854c-4b68034fd809)

If so, do you have any suggestions of the specific tooltip we could use? All the ideas coming to my mind do not seem right there - since the button already has the `Log in` label.

I would personally leave this button as-is for now - unless we come up with some specific tooltip text that fits there well. 🤔 

> Also, do you think we should add tooltips for items that are already somehow descriptive e.g. tabs, buttons like 'Add site' or 'Export entire site' etc.?

I would prefer if we keep the buttons that are descriptive enough without a tooltip. I think adding extra tooltips could make the interface too cluttered. But I am happy to discuss further.

wojtekn commented:
> If so, do you have any suggestions of the specific tooltip we could use? All the ideas coming to my mind do not seem right there - since the button already has the Log in label.

Currently, it says Log in, but it opens Settings window that let users log in. What if we added a tooltip for this one saying, 'Open settings' or 'Open settings to log in'?

> I would prefer if we keep the buttons that are descriptive enough without a tooltip. I think adding extra tooltips could make the interface too cluttered. But I am happy to discuss further.

Sounds fair, let's leave those as is for now.

wojtekn commented on src/components/header.tsx:
```diff
@@ -36,6 +43,12 @@ export default function Header() {
 							disabled={ ! site.running }
 							className="[&.is-link]:text-a8c-gray-70 [&.is-link]:hover:text-a8c-blueberry !px-0 h-0 leading-4"
 							onClick={ () => getIpcApi().openSiteURL( site.id, '', { autoLogin: false } ) }
+							label={ sprintf(
```
Should we reuse those labels for tooltip content? I've tried VoiceOver with this PR, and in those cases, when the item is active, it reads both labels one by one:

<img width="646" alt="Screenshot 2024-11-12 at 09 20 14" src="https://github.com/user-attachments/assets/74eecb26-ef71-4f6c-b2e7-29b15670e7f1">

ivan-ottinger replied:
That's a great catch, Wojtek! Thanks.

Turns out the issue is present there with existing buttons as well, e.g. `More options` or `Close`:

![Markup on 2024-11-12 at 13:56:27](https://github.com/user-attachments/assets/893eb511-5b20-4e49-beb4-62cc728aee7a)

![Markup on 2024-11-12 at 13:56:49](https://github.com/user-attachments/assets/d3293a2c-f73c-4fb7-b5e6-0c267c1d0cfa)

The root cause is in the way the  `Button`, `DropdownMenu` and `WPModal` components (defined in `@wordpress/components`) are built: As soon as the `label` property is passed to them (which is used to trigger the Tooltip), these components add `aria-label` automatically as well. Because of that, VoiceOver reads both labels instead of just one.

I have tried a couple different solutions how to address this issue. The one that seems to be the cleanest is to rely on our custom `Tooltip` component instead. In that case we wouldn't rely on `label` prop which will prevent duplicate announcements in VoiceOver and other assistive technology apps.

I have applied the changes in the latest commit (https://github.com/Automattic/studio/pull/648/commits/333b60015ff1cc5ecbc2247df947b85a727f8ed4) so you can take a closer look.

However, further workarounds would be needed to address similar existing issue with the `DropdownMenu` and `WPModal` components that handle `More options` and `Close` buttons.

Other option could be to leave things as they are for now and accept the duplication and the way the WordPress components are built.

What do you think?

ivan-ottinger replied:
Just noting that I have made a couple more changes to the branch in the meantime.

wojtekn replied:
Looks good, thanks for explaining the root case. I think we can leave the rest of components as is for now.

wojtekn reviewed (commented):
In general looks good, I spot one issue related to double labels.

ivan-ottinger committed:
Add tooltip to the `Log in` button when logged out

ivan-ottinger committed:
Use custom `Tooltip` component

ivan-ottinger committed:
Fix linting errors

ivan-ottinger committed:
Restore unintentionaly removed `showTooltip` prop from the `Button` component

ivan-ottinger committed:
Refactor `Button` component return value

wojtekn commented:
Looks good. I've only notice small bug, where Studio displays tooltip for stopped site:

<img width="295" alt="Screenshot 2024-11-13 at 08 38 35" src="https://github.com/user-attachments/assets/112a03af-d3cd-4ee1-8568-0a76a5892707">

katinthehatsite commented:
Everything looks good to me, nice work and congrats on your first Studio PR 😍 

Just to note that it also happens for the `Open site` as well: 

<img width="226" alt="Screenshot 2024-11-13 at 9 32 49 AM" src="https://github.com/user-attachments/assets/d393ff1b-6a5e-4de8-8713-698fc6352a38">

ivan-ottinger committed:
Handle header tooltip texts for stopped sites

ivan-ottinger commented:
Perfect! Thank you for catching that Wojtek and Kat!

I have addressed the issue in https://github.com/Automattic/studio/pull/648/commits/1fcd845ae408c8b99f6f9618231c233cd5849da3.

Now when the site URL is not available, the tooltips will have the following texts: `Open WP admin` and `Open site`.

katinthehatsite reviewed (approved):
Nice the changes look good to me now, thanks for moving the logic for the URL to a separate constant. It makes it much easier to read 👍 

Nice work!!

wojtekn commented:
@ivan-ottinger, It seems that after we Start and Stop the site, those two links still show the URL in the tooltip. Could we remove tooltip for those two when site is stopped, so it would be consistent with thumbnail behavior?

We will ultimately fix it when we implement https://github.com/Automattic/dotcom-forge/issues/9139/

ivan-ottinger committed:
Apply tooltip to Header links only when the site is running

ivan-ottinger commented:
> Nice the changes look good to me now, thanks for moving the logic for the URL to a separate constant. It makes it much easier to read 👍
> 
> Nice work!!

Thank you, Kat!

> @ivan-ottinger, It seems that after we Start and Stop the site, those two links still show the URL in the tooltip. Could we remove tooltip for those two when site is stopped, so it would be consistent with thumbnail behavior?
> 
> We will ultimately fix it when we implement https://github.com/Automattic/dotcom-forge/issues/9139

Sounds good! I have applied the change in https://github.com/Automattic/studio/pull/648/commits/dbf2937e8c1e6b1eab3ca591d556d8d2b86bd05b. I have also kept the fallback tooltips there in case the `site.url` isn't available for some reason.

wojtekn commented:
@ivan-ottinger I gave it a final test, and the labels look good now. However, I noticed the glitch with the help icon label:
- it's misaligned vertically
- when tooltip is shown, window gets scrollbar
- when I move cursor right, it quickly changes between both states

https://github.com/user-attachments/assets/1bdc2220-161d-4f5b-9e68-01e3373871c7

ivan-ottinger committed:
Use `title` instead of `tooltipText` for `Get help` and `Edit profile` buttons

ivan-ottinger commented:
> @ivan-ottinger I gave it a final test, and the labels look good now. However, I noticed the glitch with the help icon label:
> 
> * it's misaligned vertically
> * when tooltip is shown, window gets scrollbar
> * when I move cursor right, it quickly changes between both states
> 
>  Screen.Capture.on.2024-11-14.at.10-47-30.mp4

Thanks for catching that, Wojtek. I did not run into the same issue in my tests before but can definitely reproduce it now after resizing the app window.

Our custom `Tooltip` component is relying on an inline wrapper `div` that is causing styling conflicts in certain situations. I have now also noticed a related tiny misalignment on the `Edit profile` button in the Settings section.

In our 1:1 we were discussing the possibility of scaling down the scope of this PR and not adding the tooltip to buttons where there's some issue.

Other option could be to add the tooltip to `Get help` and `Edit profile` buttons as well, but rely on the `title` attribute there instead:

![Markup on 2024-11-14 at 15:53:01](https://github.com/user-attachments/assets/cfee84a9-d20f-42b9-9093-be274eda9427)

→ I have made the change in the last commit (https://github.com/Automattic/studio/pull/648/commits/3114ccd63ff22170f5fbebc83ff1bdce52dd4357) if you would like to try it out.

It does not look that fancy as our custom tooltip (or the other tooltip component included in WordPress), but it is more compatible and works well with accessibility tools.

I would be even willing to replace all tooltips we have with the `title` attribute until we create a tooltip solution that is compatible with all use cases.

Regardless, we can also remove tooltips from `Get help` and `Edit profile` buttons completely for now and leave only the tooltips that work well.

Do you have any preference here? 🤔

wojtekn commented:
@ivan-ottinger thanks for exploring it more. I think it would be better to skip the problematic tooltip at this point instead of having two different types of tooltip. 

I noticed one more case. Should we delete those for now, too?

<img width="675" alt="Screenshot 2024-11-15 at 09 45 00" src="https://github.com/user-attachments/assets/774bd52e-7505-42cf-8ef3-a123f8372b65">

ivan-ottinger committed:
Remove `Edit profile` tooltip

ivan-ottinger committed:
Remove `Get help` tooltip

ivan-ottinger committed:
Remove tooltips from Header links

ivan-ottinger committed:
Remove `url` from `StoppedSiteDetails` interface

ivan-ottinger commented:
> @ivan-ottinger thanks for exploring it more. I think it would be better to skip the problematic tooltip at this point instead of having two different types of tooltip.
> 
> I noticed one more case. Should we delete those for now, too?
> 
> <img alt="Screenshot 2024-11-15 at 09 45 00" width="675" src="https://private-user-images.githubusercontent.com/727413/386521006-774bd52e-7505-42cf-8ef3-a123f8372b65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2NjI1NTEsIm5iZiI6MTczMTY2MjI1MSwicGF0aCI6Ii83Mjc0MTMvMzg2NTIxMDA2LTc3NGJkNTJlLTc1MDUtNDJjZi04ZWYzLWExMjNmODM3MmI2NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNVQwOTE3MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMGUwOWIwYjVmZTZlYzQwOTkzYTExNDhlOThlNTRlODdjOThmODIzMzNjZDQxOGZjYTI0MDE1N2Q0ODAyZjE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.QrWoaCXLKtUvJseA8DF_jzVmjXBYWITI64OVHg5153Q">

Thanks, Wojtek. Sounds good to me. 👍🏼 I have removed the tooltips that were causing issues and updated the PR description accordingly.

ivan-ottinger commented:
Just noting that I have created a separate task to add the remaining tooltips: https://github.com/Automattic/dotcom-forge/issues/9841.
